app/src/main.rs

```diff
@@ -80,12 +80,52 @@ fn main(args: GlobalArgs) -> anyhow::Result<()> {
 
     // Run Context Providers
     let contexts = build_contexts(&config);
+    let can_check_for_updates = !config.disable_update_check;
+    let no_color = args.no_color;
 
     let runtime = Runtime {
         args,
         config,
         contexts,
     };
 
-    execute(runtime)
+    execute(runtime)?;
+
+    if can_check_for_updates {
```

Comment:
I would put this in right before the `execute` - block.

Reply:
👌